applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -608,6 +613,51 @@ static void send_report_mouse(void)
 	}
 }
 
+static void send_report_consumer_ctrl(void)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_HID_CONSUMER_CTRL)) {
+		struct report_data *rd =
+			&state.report_data[IN_REPORT_CONSUMER_CTRL];
+		struct hid_consumer_ctrl_event *event =
+			new_hid_consumer_ctrl_event();
+
+		event->subscriber = state.selected->id;
+		event->bitmask = 0;
+
+		/* Traverse pressed keys and build consumer control report */
+		for (size_t i = 0; i < ARRAY_SIZE(rd->items.item); i++) {
+			struct item *item = &rd->items.item[i];
+
+			if (item->value) {
+				__ASSERT_NO_MSG(item->usage_id != 0);
+				__ASSERT_NO_MSG(item->usage_id <= 16);
+
+				event->bitmask |= BIT(item->usage_id - 1);
+
+				/* Implementation assumes that RTC bits are
+				 * at the beginning of report.
+				 */
+				if (item->usage_id > CONSUMER_CTRL_REPORT_RTC_CNT) {
+					/* Removing an item and sorting array. */
+					__ASSERT_NO_MSG(rd->items.item_count != 0);
+					rd->items.item_count--;
+					item->value = 0;
+					item->usage_id = 0;
+					sort_by_usage_id(rd->items.item,
```

Comment:
sort should be done after array is traversed

Reply:
I missed that one, changing